Task 3/Task 3.1/Task 3.1/MyClosedCollection.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Task_3._1
+{
+    class MyClosedCollection<T> : IEnumerable<T>
+    {
+        private T[] _mas;
+        public MyClosedCollection()
+        {
+            Mas = Array.Empty<T>();
+        }
+        public MyClosedCollection(T[] mas)
+        {
+            Mas = mas;
+        }
+
+        public int Length { get => _mas.Length;}
+
+        public T[] Mas { get => _mas; set => _mas = value; }
```

Comment:
* Don't shorten names like that, you don't have 8 characters limit like in DOS
* You can use auto-property